ShaderTool/Command/Map.cs

```diff
@@ -337,5 +293,127 @@ class Map {
             return SUCCESS;
         }
 
+        private static int AddTexturesToResource(Stream resourceStream, MapData mapData) {
+
+            string[] mapTextureNames = mapData.textureNames;
+
+            if (mapTextureNames == null)
+                return SUCCESS;
+
+            foreach (string mapTextureName in mapTextureNames) {
+
+                string textureFilePath = Program.ResourcesFolder + "\\" + mapTextureName + ".tgx";
+
+                if (!File.Exists(textureFilePath)) {
+                    Console.WriteLine("{0} is not a texture.", mapTextureName);
+                    return WRONG_PARAMS;
+                }
+
+                byte[] textureData = File.ReadAllBytes(textureFilePath);
+                resourceStream.Write(BitConverter.GetBytes(textureData.Length));
+                resourceStream.Write(textureData);
+
+            }
+
+            return SUCCESS;
+        }
+
+        private static int AddMaterialsToResource(Stream resourceStream, MapData mapData) {
+
+            foreach (string materialName in mapData.materialNames) {
+
+                // Material not found
+                if (!Cache.MATERIALS.ContainsKey(materialName)
+                  || Cache.MATERIALS[materialName] == null) { // Not sure if this null check is even necessary, but better safe than sorry
+
+                    Console.WriteLine("Material {0} was not found!", materialName);
+                    return WRONG_PARAMS;
+
+                }
+
+                MaterialData materialData = Cache.MATERIALS[materialName];
+
+                // Write color to file, provided as an R,G,B,A float array
+                if (materialData.color == null)
+                    for (int i = 0; i < 4; i++)
+                        resourceStream.Write(BitConverter.GetBytes(0f));
+                else
+                    foreach (float value in materialData.color)
+                        resourceStream.Write(BitConverter.GetBytes(value));
+
+                // Write diffuse texture to file
+                resourceStream.Write(BitConverter.GetBytes(materialData.diffuseTexture));
+
+            }
+
+            return SUCCESS;
+
+        }
+
+        private static int AddActorsToResource(Stream resourceStream, MapData mapData) {
+
+            string[] allMaterialNames = Cache.MATERIALS.Keys.ToArray();
+
+            // Write the data names into the resource file
+            foreach (string actorName in mapData.actorNames) {
+
+                string actorFilePath = Program.ResourcesFolder + @"\" + actorName + "_Actor.json";
+
+                if (!File.Exists(actorFilePath)) {
+                    Console.WriteLine("{0} is not a valid actor!", actorName);
+                    return WRONG_PARAMS;
+                }
+
+                ActorData actorData = JsonConvert.DeserializeObject<ActorData>(File.ReadAllText(actorFilePath));
+
```

Comment:
I even forgot to implement the sizes
```suggestion
resourceStream.Write(BitConverter.GetBytes(26 + (actorData.indexCount + actorData.vertexCount) * 4)); // 2 byte + 2(4 +8)
```